mobile/src/bundles/auth/store/actions.ts

```diff
@@ -16,9 +17,21 @@ const signUp = createAsyncThunk<
     UserSignUpResponseDto,
     UserSignUpRequestDto,
     AsyncThunkConfig
->(`${sliceName}/sign-up`, (signUpPayload, { extra }) => {
-    const { authApi } = extra;
-    return authApi.signUp(signUpPayload);
+>(`${sliceName}/sign-up`, async (signUpPayload, { extra }) => {
+    const { authApi, storage } = extra;
+    try {
+        const response = await authApi.signUp(signUpPayload);
+        await storage.set(StorageKey.TOKEN, response.token);
+
+        return response;
+    } catch (error) {
+        if (error instanceof Error) {
+            notifications.showError(error.message);
+            throw error;
+        }
+        notifications.showError(ErrorMessages.UNKNOWN_ERROR);
```

Comment:
Let's move this check to some helper, we'll have a lot of code duplication in the future I guess. Let's create some helper which will return us an error message, and we'll send notification with this message